test/support/http_case.ex

```diff
@@ -1,40 +1,54 @@
 defmodule Castle.HttpCase do
   use ExUnit.CaseTemplate, async: false
 
-  def resp("" <> body, code) do
+  def mock_http_resp({code, "" <> body}) do
     {:ok, %HTTPoison.Response{status_code: code, body: body}}
   end
-  def resp(body, code) do
+  def mock_http_resp({code, body}) do
     {:ok, encoded_body} = Poison.encode(body)
-    resp(encoded_body, code)
+    mock_http_resp({code, encoded_body})
   end
-  def resp(body), do: resp(body, 200)
+  def mock_http_resp(code) when is_integer(code), do: mock_http_resp({code, ""})
+  def mock_http_resp(body), do: mock_http_resp({200, body})
 
-  using do
-    quote do
-      import Mock
+  def mock_full_url(url, nil), do: url
+  def mock_full_url(url, opts) do
+    case opts[:params] do
+      nil -> url
+      params when map_size(params) == 0 -> url
+      params -> "#{url}?#{URI.encode_query(params)}"
+    end
+  end
 
-      defmacro with_http(body, do: expression) do
-        quote do
-          getter = fn(_url, _hdrs, _opts) ->
-            unquote(body) |> Castle.HttpCase.resp()
-          end
-          with_mock HTTPoison, [get: getter] do
-            unquote(expression)
+  defmacro test_with_http(name, mocks, do: expression) do
```

Comment:
A bit hard to read, but this makes more sense if you look at how it's used in the tests.